cpp/arcticdb/entity/descriptor_item.hpp

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <arcticdb/entity/atom_key.hpp>
+#include <fmt/format.h>
+#include <string>
+
+
+namespace arcticdb {
+struct DescriptorItem {
+    DescriptorItem(entity::AtomKey &&key, std::optional<IndexValue> start_index, std::optional<IndexValue> end_index) :
+        key_(std::move(key)),
+        start_index_(start_index),
+        end_index_(end_index) {
+    }
+
+    DescriptorItem(entity::AtomKey &key, std::optional<IndexValue> start_index, std::optional<IndexValue> end_index) :
```

Comment:
Do we really need both of these constructors?

Reply:
Discussed in person, should only need the && version. Remove this ctor

Reply:
done